src/components/ReadStats.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import {
+  Modal,
+  ModalBody
+} from 'reactstrap';
+import { connect } from 'react-redux';
+import ArticleListCard from './common/ArticleList';
+import CancelIcon from './common/CancelIcon';
+import { closeReadStatsModal } from '../actions/profileActions';
+
+class ReadStats extends Component {
+  closeReadModal = () => {
+    const { dispatch } = this.props;
+    dispatch(closeReadStatsModal());
+  }
+
+  render() {
+    const { isReadStatsModalOpen, readList } = this.props;
+    return (
+      <div className='readstats'>
+        <Modal isOpen={isReadStatsModalOpen} className='readstats-body'>
+          <CancelIcon clicked={this.closeReadModal} />
+          {readList.length === 0
+            ? <h5 className='readstats-h5'>No Articles Read Yet</h5>
+            : <h5 className='readstats-h5'>Articles Read</h5>}
+          <br />
+          <ModalBody className='readstats-body'>
+            {
+                readList.map(article => (
+                  <ArticleListCard
+                    key={Math.random(Date.now())}
```

Comment:
The key you set to random here is going to be changing every second, which means that react will find it difficult to keep track of which one should be re-rendered when state changes. 
see [here](https://reactjs.org/docs/lists-and-keys.html#keys) and [here](https://reactjs.org/docs/reconciliation.html#recursing-on-children)

Reply:
> The key you set to random here is going to be changing every second, which means that react will find it difficult to keep track of which one should be re-rendered when state changes.
> see [here](https://reactjs.org/docs/lists-and-keys.html#keys) and [here](https://reactjs.org/docs/reconciliation.html#recursing-on-children)
> 
> Just passing :)

Math.random(Date.now()) is a function that returns a value when initialized and based on the code, the map value only renders once in the DOM provided there is no data mutation which means react will b able to identify each iteration element based on the unique Math.randon(Date.now()) return value.